src/components/ImageView/index.spec.tsx

```diff
@@ -55,4 +57,22 @@ describe(__filename, () => {
     expect(_btoa).toHaveBeenCalled();
     expect(_log.debug).toHaveBeenCalled();
   });
+
+  it('calls sanitize on svg files', () => {
+    const _sanitize = jest.fn();
+    const content = 'some content';
+    const mimeType = 'image/svg+xml';
+    render({ _sanitize, content, mimeType });
```

Comment:
I think we should add an additional test to cover the integration of all parts. I'm mostly concerned about this because the test I'm commenting on doesn't cover the actual `src` value.

This is a bit paranoid and usually I don't like to test functionality of third party libraries but I think this could go a long way to ensure everything is hooked up correctly:

```diff
diff --git a/src/components/ImageView/index.spec.tsx b/src/components/ImageView/index.spec.tsx
index 0ff5fd2..fb48bb0 100644
--- a/src/components/ImageView/index.spec.tsx
+++ b/src/components/ImageView/index.spec.tsx
@@ -3,23 +3,23 @@ import { shallow } from 'enzyme';
 
 import { createFakeLogger } from '../../test-helpers';
 
-import ImageView from '.';
+import ImageView, { PublicProps } from '.';
 
 describe(__filename, () => {
   const render = ({
     _btoa = btoa,
     _log = createFakeLogger(),
-    _sanitize = jest.fn(),
     content = 'some image content',
     mimeType = 'mime/type',
-  } = {}) => {
+    ...props
+  }: Partial<PublicProps> = {}) => {
     return shallow(
       <ImageView
         _btoa={_btoa}
         _log={_log}
-        _sanitize={_sanitize}
         mimeType={mimeType}
         content={content}
+        {...props}
       />,
     );
   };
@@ -75,4 +75,24 @@ describe(__filename, () => {
 
     expect(_sanitize).not.toHaveBeenCalled();
   });
+
+  it('sanitizes an svg file', () => {
+    const content = `
+      <svg version="1.1">
+        <circle fill="red" />
+        <script>alert('XSS via SVG')</script>
+      </svg>
+    `;
+    const root = render({ content, mimeType: 'image/svg+xml' });
+
+    const parts = root
+      .find('img')
+      .prop('src')
+      .split(',');
+    const contentPart = parts[parts.length - 1];
+
+    const svg = String(atob(contentPart));
+    expect(svg).toContain('<circle');
+    expect(svg).not.toContain('<script>');
+  });
 });
diff --git a/src/components/ImageView/index.tsx b/src/components/ImageView/index.tsx
index 971d781..93c040f 100644
--- a/src/components/ImageView/index.tsx
+++ b/src/components/ImageView/index.tsx
@@ -4,7 +4,7 @@ import * as React from 'react';
 
 import styles from './styles.module.scss';
 
-type PublicProps = {
+export type PublicProps = {
   _btoa?: typeof btoa;
   _log?: typeof log;
   _sanitize?: typeof purify.sanitize;

```

Reply:
Yes, this seems like a good test. I'll add it.